core/domain/exp_jobs_one_off.py

```diff
@@ -128,19 +133,32 @@ def entity_classes_to_map_over(cls):
 
     @staticmethod
     def map(item):
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+        ltt = latex2text.LatexNodes2Text()
+        valid_exps_seen = 0
+
         if not item.deleted:
             exploration = exp_fetchers.get_exploration_from_model(item)
             for state_name, state in exploration.states.items():
                 if state.interaction.id == 'MathExpressionInput':
                     for group in state.interaction.answer_groups:
                         for rule_spec in group.rule_specs:
-                            rule = rule_spec.inputs['x']
-                            if any(sep in rule for sep in SEPARATORS):
+                            rule_input = ltt.latex_to_text(
+                                rule_spec.inputs['x'])
+                            validity = 'Invalid'
+                            if is_valid_math_expression(rule_input):
+                                validity = 'Valid Expression'
+                                valid_exps_seen += 1
+                            elif is_valid_math_equation(rule_input):
+                                validity = 'Valid Equation'
+                                valid_exps_seen += 1
+                            if valid_exps_seen <= MATH_VALID_EXP_LIMIT:
```

Comment:
You aren't going to reach this limit in a single exploration. The idea was to cap the output, so the limit should be applied in the reduce() step.

For invalid stuff, presumably you will want to include more debug info so that the issues can be fixed manually if necessary. E.g. the exp ID and some info about where the problematic expression was found.

Reply:
> the limit should be applied in the reduce() step.

@seanlip Just to confirm, we would then have to use a class variable like I'd done before, right? If I'm not mistaken, static methods (map and reduce) won't be able to access instance variables (if I were to initialize it in the `__init__` method).